src/Microsoft.ML.AutoML/AutoMLExperiment/AutoMLExperiment.cs

```diff
@@ -40,31 +41,57 @@ public AutoMLExperiment(MLContext context, AutoMLExperimentSettings settings)
             }
 
             _serviceCollection = new ServiceCollection();
+            InitializeServiceCollection();
         }
 
         private void InitializeServiceCollection()
         {
             _serviceCollection.TryAddTransient((provider) =>
             {
+                var channel = provider.GetRequiredService<IChannel>();
                 var context = new MLContext(_settings.Seed);
-                _cts.Token.Register(() =>
+                _globalCancellationTokenSource.Token.Register(() =>
                 {
                     // only force-canceling running trials when there's completed trials.
                     // otherwise, wait for the current running trial to be completed.
                     if (_bestTrialResult != null)
                         context.CancelExecution();
                 });
 
+                context.Log += (o, e) =>
+                {
+                    // Relay logs that are generated by the current MLContext to the Experiment class's
+                    // _logger.
+                    switch (e.Kind)
+                    {
+                        case ChannelMessageKind.Trace:
+                            channel.Trace(e.Message);
+                            break;
+                        case ChannelMessageKind.Info:
+                            channel.Info(e.Message);
+                            break;
+                        case ChannelMessageKind.Warning:
+                            channel.Warning(e.Message);
+                            break;
+                        case ChannelMessageKind.Error:
+                            channel.Error(e.Message);
+                            break;
+                        default:
+                            break;
+                    }
+                };
+
                 return context;
             });
+
+            this.SetPerformanceMonitor(2000);
             _serviceCollection.TryAddSingleton(_settings);
             _serviceCollection.TryAddSingleton(((IChannelProvider)_context).Start(nameof(AutoMLExperiment)));
         }
 
         private void Initialize()
```

Comment:
>Initialize [](http://example.com/codeflow?start=21&length=10)

nit: maybe getting rid of this method be better? it has now only one statement.

Reply:
### Resolved